### PR TITLE
Game data wrappers and dev tool build configurations w/ positional radar

### PR DESCRIPTION
**Pull request by Mogster7**
This is primarily a feature push that was a long time coming.

Includes:

- Game data containers. Found in GameData.h/cpp, these are used to query PlayerControl/PlayerInfo for internal data and are not reliant on RPC packets at any stage. 
- Dev tool build configurations. Called **RelWithDevTools**, this provides the preprocessor definition of 'DEV_TOOLS', otherwise being identical to the release build. This allows us to build with debugging features that'd typically be unavailable to the end-user. The build configurations have been updated to post-build dump **their** winhttp.dll to the Among Us steam folder, as opposed to strictly the Release variant.
- Positional Radar dev tool, integrated into ImGui. This still needs some work for ghost mode and the render can be improved by walls and such. 
-  The functions for the above features (along with some deprecated ones that I left in case anybody needed them) have been included in the deobfuscation header, and some have been added to the MatcherDefinitions cs file for scalability where possible. The deobfuscation header now also has a newly created CPP file for its definitions, as to avoid redefinition errors if referenced in multiple areas.

### CODE REVIEW

File: AUMInjector/deobfuscate/2020_12_9s.cpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include <deobfuscate.h>
```
Comment (StarGate01) [inline AUMInjector/deobfuscate/2020_12_9s.cpp:1]: Why did you move the enum definitions to a seperate cpp file?

File: AUMInjector/deobfuscate/2020_12_9s.h [hunk 2]
```diff
@@ -1,20 +1,45 @@
 using Player_Die_Reason__Enum = DBLJKMDLJIF__Enum;
```
Comment (StarGate01) [inline AUMInjector/deobfuscate/2020_12_9s.h:1]: `#pragma once` should take care of doubel definition errors

File: AUMInjector/user/GameData.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#pragma once
```
Comment (StarGate01) [inline AUMInjector/user/GameData.h:1]: I like this object-orientated approach.

**Review by StarGate01 — COMMENTED** [review]
I added a few comments.

I like the usage of data classes and namespaces. We should eventually adjust the existing code (especially `MumblePlayer`) to fit with the namespaces.

File: AUMInjector/deobfuscate/2020_12_9s.cpp (hunk 1 above)
Comment (Mogster7) [reply AUMInjector/deobfuscate/2020_12_9s.cpp:1]: This was a fix to an issue I was experiencing where I was getting redefinition errors on including the deobfuscation header in both main.cpp **and** GameData.cpp. The pragma once for whatever reason didn't seem to fix it, but I can give it a double-check.

Comment (Mogster7) [reply AUMInjector/deobfuscate/2020_12_9s.cpp:1]: image

This was with a `#pragma once` at the top of the header.
After getting some advice from @ShumWengSang, it looks like the solutions to this problem are moving it to the cpp, using the inline keyword (C++17 and up) or possibly including the deobfuscation header in the PCH for compilation only once?

How would you like to proceed?

Comment (StarGate01) [reply AUMInjector/deobfuscate/2020_12_9s.cpp:1]: I see. I'll merge it and test the rest of the code. `__declspec( selectany )` seems to be a promising MSVC alternative to `inline` for older C++ versions.
